File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/KeyboardInitializer.java
package com.websarva.wings.android.zuboradiary.ui;

import android.app.Activity;
import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

import java.util.Objects;

public class KeyboardInitializer {

    private final InputMethodManager inputMethodManager;

    public KeyboardInitializer(Activity activity) {
        Objects.requireNonNull(activity);

        inputMethodManager =
                (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
    }

    public void hide(View focusView) {
        Objects.requireNonNull(focusView);

        inputMethodManager
                .hideSoftInputFromWindow(focusView.getWindowToken(), InputMethodManager.HIDE_NOT_ALWAYS);
    }

    public void show(View focusView) {
        Objects.requireNonNull(focusView);

        inputMethodManager
                .showSoftInput(focusView, InputMethodManager.RESULT_UNCHANGED_SHOWN);
    }
}

File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/KeyboardInitializer.kt
package com.websarva.wings.android.zuboradiary.ui

import android.app.Activity
import android.content.Context
import android.util.Log
import android.view.View
import android.view.inputmethod.InputMethodManager

internal class KeyboardInitializer(activity: Activity) {
    private val inputMethodManager: InputMethodManager =
        activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager

    fun hide(focusView: View) {
        Log.d(javaClass.simpleName, "hide()")
        inputMethodManager
            .hideSoftInputFromWindow(focusView.windowToken, InputMethodManager.HIDE_NOT_ALWAYS)
    }

    fun show(focusView: View) {
        Log.d(javaClass.simpleName, "show()")
        inputMethodManager.showSoftInput(focusView, InputMethodManager.SHOW_IMPLICIT)
    }
}
